src/app/ngrx/reducers/selectors/downloads-table.selectors.ts

```diff
@@ -21,31 +23,39 @@ export const selectDownloadTablePodcastDownloads = createSelector(
    podcastAllTimeDownloads: PodcastAllTimeDownloads): DownloadsTableModel => {
     let podcastData: DownloadsTableModel;
 
-    if (podcastDownloads && routerParams.chartType !== CHARTTYPE_EPISODES) {
-      const data = podcastDownloads.downloads;
-      if (data) {
-        const totalForPeriod = getTotal(data);
-        podcastData = {
-          id: routerParams.podcastId,
-          title: 'All Episodes',
-          downloads: mapMetricsToTimeseriesData(data),
-          color: routerParams.chartType === CHARTTYPE_PODCAST ? standardColor : neutralColor,
-          totalForPeriod,
-          charted: podcastDownloads.charted
-        };
-        if (podcastAllTimeDownloads) {
-          if (totalForPeriod > podcastAllTimeDownloads.allTimeDownloads) {
-            podcastData.allTimeDownloads = totalForPeriod;
-          } else {
-            podcastData.allTimeDownloads = podcastAllTimeDownloads.allTimeDownloads;
-          }
+    if (podcastDownloads && routerParams.chartType !== CHARTTYPE_EPISODES && podcastDownloads.downloads) {
+      const totalForPeriod = getTotal(podcastDownloads.downloads);
+      podcastData = {
+        id: routerParams.podcastId,
+        title: 'All Episodes',
+        color: routerParams.chartType === CHARTTYPE_PODCAST ? standardColor : neutralColor,
+        totalForPeriod,
+        charted: podcastDownloads.charted
+      };
+      if (podcastAllTimeDownloads) {
+        if (totalForPeriod > podcastAllTimeDownloads.allTimeDownloads) {
+          podcastData.allTimeDownloads = totalForPeriod;
+        } else {
+          podcastData.allTimeDownloads = podcastAllTimeDownloads.allTimeDownloads;
         }
       }
     }
     return podcastData;
   }
 );
 
```

Comment:
The downloads array has a separate selector now. Building this any[][] with the formatted headers in the selector is better for performance. Previously these dates were formatted in the component and checked on change detection.